android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/BridgeManagerProvider.java

```diff
@@ -1,52 +1,157 @@
+import org.sagebionetworks.bridge.android.data.StudyUploadEncryptor;
+import org.sagebionetworks.bridge.android.manager.dao.AccountDAO;
+import org.sagebionetworks.bridge.android.manager.dao.ConsentDAO;
+import org.sagebionetworks.bridge.rest.ApiClientProvider;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
 
 /**
- * Created by jyliu on 1/20/2017.
+ * This class contains configuration and performs setup of dependencies for a Bridge Application.
+ * <p>
+ * A call to {@link #getInstance()} returns a BridgeManagerProvider instance that is usable
+ * globally. The managers made available by BridgeManagerProvider maintain and share state across
+ * the Application.
  */
 @AnyThread
 public class BridgeManagerProvider {
     private static BridgeManagerProvider instance;
 
+    /**
+     * @return singleton instance
+     */
     @NonNull
-    public static BridgeManagerProvider getInstance(@NonNull Context context) {
-        if (instance == null) {
-            instance = new BridgeManagerProvider(new BridgeConfig(context));
-        }
+    public static BridgeManagerProvider getInstance() {
+        checkState(instance != null, "BridgeManagerProvider has not been initialized. " +
+                "Call init(Context) in your Application#onCreate()");
 
         return instance;
     }
 
+    /**
+     * Allows injection of singleton instance for testing purposes.
+     *
+     * @param bridgeManagerProvider instance to be returned by {@link #getInstance()}
+     */
+    public static void init(@NonNull BridgeManagerProvider bridgeManagerProvider) {
+        checkNotNull(bridgeManagerProvider);
+        instance = bridgeManagerProvider;
+    }
+
+    /**
+     * Intended to be called in {@link Application#onCreate()}
+     *
+     * @param applicationContext application's global context
+     */
+    public static void init(@NonNull Context applicationContext) {
+        checkNotNull(applicationContext);
+        checkState(instance == null, "BridgeManagerProvider has already been initialized");
+
+        instance = new BridgeManagerProvider(applicationContext);
+    }
+
+    private BridgeManagerProvider(@NonNull Context applicationContext) {
+        this.applicationContext = applicationContext;
+        bridgeConfig = new BridgeConfig(this.applicationContext);
+
+        apiClientProvider = new ApiClientProvider(
+                bridgeConfig.getBaseUrl(),
+                bridgeConfig.getUserAgent(),
+                bridgeConfig.getAcceptLanguage());
+
+        accountDAO = new AccountDAO(applicationContext);
+        consentDAO = new ConsentDAO(applicationContext);
+
+        authenticationManager = new AuthenticationManager(bridgeConfig, apiClientProvider, accountDAO);
+        participantManager = new ParticipantManager(authenticationManager, accountDAO);
+        consentManager = new ConsentManager(authenticationManager, consentDAO);
+
+        try {
```

Comment:
We went away from lazy loading, so I'll point out this does load from assets here (perhaps that's non-trivial work?) My intention is that this will crash the app early if there will be a problem encrypting data archives so I'm wrapping in an unchecked exception and rethrowing. StudyUploadEncryptor gives a meaningful error message, hence the lame one here.